### PR TITLE
Improve login.jsp file to handle custom local authenticators

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Purpose
The custom local authenticators can be registered as an extension point and can be added to the applications.
When there are multiple options for the application, it doesn't show the custom local authenticators as the existing login.jsp doesn't have that support as it only support the already registered defined local authenticators.

This PR is initiated to provide that support, and the documentation should be improved to guide the developers to provide the custom local authenticator logo image and i18 configs as necessary.

### Related Issues
- https://github.com/wso2/product-is/issues/20265

### After the fix.

Screenshot 2024-09-12 at 09 56 16

### CODE REVIEW

File: identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp [hunk 1]
```diff
@@ -170,6 +170,8 @@
     boolean hasLocalLoginOptions = false;
     boolean isBackChannelBasicAuth = false;
     List<String> localAuthenticatorNames = new ArrayList<String>();
+    List<String> registeredLocalAuthenticators = Arrays.asList(BACKUP_CODE_AUTHENTICATOR,TOTP_AUTHENTICATOR,EMAIL_OTP_AUTHENTICATOR,MAGIC_LINK_AUTHENTICATOR,SMS_OTP_AUTHENTICATOR,OPEN_ID_AUTHENTICATOR,IDENTIFIER_EXECUTOR,JWT_BASIC_AUTHENTICATOR,BASIC_AUTHENTICATOR,IWA_AUTHENTICATOR,X509_CERTIFICATE_AUTHENTICATOR,FIDO_AUTHENTICATOR);
```
Comment (pavinduLakshan) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp:173]: Shall we span the authenticator list across multiple lines for better readability?

```suggestion
    List<String> registeredLocalAuthenticators = Arrays.asList(   
        BACKUP_CODE_AUTHENTICATOR, TOTP_AUTHENTICATOR, EMAIL_OTP_AUTHENTICATOR,
        MAGIC_LINK_AUTHENTICATOR,SMS_OTP_AUTHENTICATOR,OPEN_ID_AUTHENTICATOR,
        IDENTIFIER_EXECUTOR,JWT_BASIC_AUTHENTICATOR,BASIC_AUTHENTICATOR,
        IWA_AUTHENTICATOR,X509_CERTIFICATE_AUTHENTICATOR,FIDO_AUTHENTICATOR
   );
```

File: identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp [hunk 2]
```diff
@@ -1052,6 +1059,37 @@
...
+                                            alt="<%=localAuthenticator%> Logo"
+                                            role="presentation">
+                                        <span>
+                                            <%=AuthenticationEndpointUtil.i18n(resourceBundle, "sign.in.with")%>
+                                            <%=AuthenticationEndpointUtil.i18n(resourceBundle, localAuthenticator)%>
+                                        </span>
+                                        </button>
+                                </div>
+                            </div>
+                            <br>
```
Comment (DilshanSenarath) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp:1090]:
```suggestion
                              <div class="social-login blurring social-dimmer">
                                    <div class="field">
                                            <button
                                                type="button"
                                                id="icon-<%=iconId%>"
                                                class="ui button secondary"
                                                data-testid="login-page-sign-in-with-<%=localAuthenticator%>"
                                                onclick="handleNoDomain(this,
                                                        '<%=Encode.forJavaScriptAttribute(Encode.forUriComponent(idpEntry.getKey()))%>',
                                                        '<%=localAuthenticator%>')"
                                            >
                                                <img
                                                    class="ui image"
                                                    src="libs/themes/default/assets/images/authenticators/<%=localAuthenticator%>.svg"
                                                    alt="<%=localAuthenticator%> Logo"
                                                    role="presentation">
                                                <span>
                                                    <%=AuthenticationEndpointUtil.i18n(resourceBundle, "sign.in.with")%>
                                                    <%=AuthenticationEndpointUtil.i18n(resourceBundle, localAuthenticator)%>
                                                </span>
                                            </button>
                                    </div>
                            </div>
                            <br>
```

Comment (DilshanSenarath) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp:1090]: If we can let's format the div elements correctly

**Review by DilshanSenarath — APPROVED** [review]

File: identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp (hunk 2 above)
Comment (sadilchamishka) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp]: Fixed by - https://github.com/wso2/identity-apps/pull/6889/commits/8f8f8906694e045debe1f27475a21b04be0c8d9a

**Review by DilshanSenarath — APPROVED** [review]

**Pull request merged**